ui/src/component/waterfall.js

```diff
@@ -0,0 +1,126 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import ReactEcharts from 'echarts-for-react';
+import _ from 'lodash';
+import { validate } from '../utils';
+
+export default class Sankey extends PureComponent {
+  static propTypes = {
+    source: PropTypes.arrayOf(PropTypes.array).isRequired,
+  }
+
+  render() {
+    const { source } = this.props;
+    validate(source);
+
+    const columns = _.zip(...source);
+    const option = {
+      tooltip: {
+        trigger: 'axis',
+        axisPointer: {
+          type: 'shadow',
+        },
+        formatter(params) {
+          let target;
+          if (_.get(params, '[1].value') !== '-') {
+            target = _.get(params, 1);
+          } else {
+            target = _.get(params, 2);
+          }
+          return `${target.name}<br/>${target.seriesName} : ${target.value}`;
+        },
+      },
+      legend: {
+        show: false,
+      },
+      xAxis: [
+        {
+          type: 'category',
+          splitLine: { show: false },
+          data: _.chain(columns).first().slice(1).value(),
+        },
+      ],
+      yAxis: [
+        {
+          type: 'value',
+        },
+      ],
+      series: (() => {
+        const assistant = [];
+        const positive = [];
+        const negtive = [];
+        //               = assistant[i-1]        (num >= 0)
```

Comment:
The description doesn't seem correct to me.

Reply:
Fixed